tests/api/test_statements_post.py

```diff
@@ -0,0 +1,229 @@
+                    "scopes": ["ralph_test_scope"],
+                }
+            ]
+        ),
+    )
+
+    return credentials
+
+
+@pytest.fixture(autouse=True)
+def teardown_es_index():
+    """
+    Clean up Elasticsearch after each test.
+    """
+    ES_INDICES_CLIENT.create(ES_TEST_INDEX)
+    yield
+    ES_INDICES_CLIENT.delete(ES_TEST_INDEX)
+
+
+# pylint: disable=invalid-name
+def test_post_statements_list_of_one(fs):
+    """
+    Statements POST supports adding one statement by just
+    passing a list of one statement.
+    """
+    credentials = setup_auth(fs)
+
+    statement = {
+        "actor": {
+            "account": {
+                "homePage": "https://example.com/homepage/",
+                "name": str(uuid4()),
+            },
+            "objectType": "Agent",
+        },
+        "id": str(uuid4()),
+        "object": {"id": "https://example.com/object-id/1/"},
+        "timestamp": 1647349671,
+        "verb": {"id": "https://example.com/verb-id/1/"},
+    }
+
+    response = client.post(
+        "/xAPI/statements/",
+        headers={"Authorization": f"Basic {credentials}"},
+        json=[statement],
+    )
+
+    assert response.status_code == 200
+    assert response.json() == [statement["id"]]
+    ES_INDICES_CLIENT.refresh()
+    hits = ES_CLIENT.search(index="statements")["hits"]["hits"]
+    assert len(hits) == 1
+    assert hits[0]["_id"] == statement["id"]
+
+
+# pylint: disable=invalid-name
+def test_post_statements_list(fs):
+    """
+    Statements POST supports adding more than one statement by just
+    passing a list of statements.
+    """
+    credentials = setup_auth(fs)
+
+    statements = [
+        {
+            "actor": {
+                "account": {
+                    "homePage": "https://example.com/homepage/",
+                    "name": str(uuid4()),
+                },
+                "objectType": "Agent",
+            },
+            "id": str(uuid4()),
+            "object": {"id": "https://example.com/object-id/1/"},
+            "timestamp": 1647349671,
+            "verb": {"id": "https://example.com/verb-id/1/"},
+        },
+        {
+            "actor": {
+                "account": {
+                    "homePage": "https://example.com/homepage/",
+                    "name": str(uuid4()),
+                },
+                "objectType": "Agent",
+            },
+            # Note the second statement has no preexisting ID
+            "object": {"id": "https://example.com/object-id/1/"},
+            "timestamp": 1647349671,
+            "verb": {"id": "https://example.com/verb-id/1/"},
+        },
+    ]
+
+    response = client.post(
+        "/xAPI/statements/",
+        headers={"Authorization": f"Basic {credentials}"},
+        json=statements,
+    )
+
+    assert response.status_code == 200
+    assert response.json()[0] == statements[0]["id"]
+    assert isinstance(response.json()[1], str)
```

Comment:
Do we know if it is only a random string ? Or is it an UUID?

Reply:
It should be a UUID in string format. Maybe you're suggesting we check that with a regex?

Reply:
Yup, it should be!